### ISSUE TITLE
git.latest identity does not set the correct user for the private key file on the minion

### ISSUE THREAD

**Issue by michaelkarrer81** [saltstack/salt#41936]
### Description of Issue/Question
Wrong linux user on the minion for the identity file if an salt:// path is used! 

```
git.latest:
    - name: git@github.com:Test/test.git
    - rev: master
    - target: /opt/mygitrepo
    - branch: master
    - user: mike
    - identity: salt://ssh/deploy/deploy_id_rsa      # Private ssh key for github deploy key
    - fetch_tags: True
    - force_checkout: True
    - force_clone: True
    - force_fetch: True
    - force_reset: True
```

Will load the private key file to the minion **but owned by root** and **not** by linux user **mike**!
This leads to an permission error later because linux user mike can not read the private key file! 

### Versions Report
Salt Version:
           Salt: 2016.11.3
 
Dependency Versions:
           cffi: 1.5.0
       cherrypy: 3.2.3
       dateutil: 1.5
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: 0.23.3
        libnacl: 1.4.4
       M2Crypto: Not Installed
           Mako: 0.9.1
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: 1.2.3
      pycparser: 2.14
       pycrypto: 2.6.1
         pygit2: 0.23.1
         Python: 2.7.6 (default, Oct 26 2016, 20:30:19)
   python-gnupg: Not Installed
         PyYAML: 3.10
          PyZMQ: 14.0.1
           RAET: Not Installed
          smmap: 0.8.2
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
 
System Versions:
           dist: Ubuntu 14.04 trusty
        machine: x86_64
        release: 3.13.0-32-generic
         system: Linux
        version: Ubuntu 14.04 trusty

**Comment by garethgreenaway** [comment]
Thanks for the report @michaelkarrer81

**Comment by garethgreenaway** [comment]
This issue should be considered resolved and the fix is available on 2016.11. If the issue is not considered resolved or persists, please feel free to re-open this issue.
